hybridse/src/vm/engine.cc

```diff
@@ -462,17 +462,23 @@ int32_t RequestRunSession::Run(const uint32_t task_id, const Row& in_row, Row* o
     if (!sql_request_rows.empty()) {
         row = sql_request_rows.at(0);
     }
-    auto task = std::dynamic_pointer_cast<SqlCompileInfo>(compile_info_)
-                    ->get_sql_context()
-                    .cluster_job->GetTask(task_id)
-                    .GetRoot();
+
+    auto info = std::dynamic_pointer_cast<SqlCompileInfo>(compile_info_);
+    if (!info->GetRequestSchema().empty() && row.empty()) {
```

Comment:
some SubQuery may input request row as empty: a empty to represent a row with one NULL value. 

That's why `EngineTestLastJoinComplex/EngineTest.TestClusterRequestEngine/9` fails

```
SIMPLE_PROJECT(sources=(auto_t0.id, auto_t0.c1, auto_t0.c3, auto_t1.c3, auto_t2.c4))                                                                                                                                            
  REQUEST_JOIN(type=kJoinTypeConcat)                                                                                                                                                                                            
    REQUEST_JOIN(type=kJoinTypeConcat)                                                                                                                                                                                          
      DATA_PROVIDER(request=auto_t0)                                                                                                                                                                                            
      REQUEST_JOIN(OUTPUT_RIGHT_ONLY, type=LastJoin, right_sort=(ASC), condition=, left_keys=(), right_keys=(), index_keys=(#10))                                                                                               
        SIMPLE_PROJECT(sources=(#10 -> auto_t0.c1))                                                                                                                                                                             
          DATA_PROVIDER(request=auto_t0)                                                                                                                                                                                        
        DATA_PROVIDER(type=Partition, table=auto_t1, index=index1)                                                                                                                                                              
    REQUEST_JOIN(OUTPUT_RIGHT_ONLY, type=LastJoin, right_sort=(ASC), condition=, left_keys=(), right_keys=(), index_keys=(#171))                                                                                                
      SIMPLE_PROJECT(sources=(#171 -> auto_t1.c3))                                                                                                                                                                              
        REQUEST_JOIN(OUTPUT_RIGHT_ONLY, type=LastJoin, right_sort=(ASC), condition=, left_keys=(), right_keys=(), index_keys=(#10))                                                                                             
          SIMPLE_PROJECT(sources=(#10 -> auto_t0.c1))                                                                                                                                                                           
            DATA_PROVIDER(request=auto_t0)                                                                                                                                                                                      
          DATA_PROVIDER(type=Partition, table=auto_t1, index=index1)                                                                                                                                                            
      DATA_PROVIDER(type=Partition, table=auto_t2, index=index2)                                                                                                                                                                
I20240511 04:39:54.296554  6077 engine_test_base.cc:434] Runner plan:                                                                                                                                                           
TASK ID 0, routing index = test_zw.auto_t1.index1, keys=(#10)                                                                                                                                                                   
[3]REQUEST_JOIN OUTPUT_RIGHT_ONLY (cache_enable)                                                                                                                                                                                
  [4]REQUEST                                                                                                                                                                                                                    
  [2]DATA                                                                                                                                                                                                                       
TASK ID 1, routing index = test_zw.auto_t2.index2, keys=(#171)                                                                                                                                                                  
[9]REQUEST_JOIN OUTPUT_RIGHT_ONLY                                                                                                                                                                                               
  [10]REQUEST                                                                                                                                                                                                                   
  [8]DATA                                                                                                                                                                                                                       
TASK ID 2, routing index = test_zw.auto_t1.index1, keys=(#10)                                                                                                                                                                   
[11]REQUEST_RUN_PROXY(TASK_ID=1) lazy                                                                                                                                                                                           
  [7]SIMPLE_PROJECT lazy                                                                                                                                                                                                        
    [3]REQUEST_JOIN OUTPUT_RIGHT_ONLY (cache_enable)                                                                                                                                                                            
      [4]REQUEST                                                                                                                                                                                                                
      [2]DATA                                                                                                                                                                                                                   
MAIN TASK ID 3                                                                                                                                                                                                                  
[14]SIMPLE_PROJECT lazy                                                                                                                                                                                                         
  [12]CONCAT lazy                                                                                                                                                                                                               
    [5]CONCAT lazy                                                                                                                                                                                                              
      [0]REQUEST (cache_enable)                                                                                                                                                                                                 
      [6]REQUEST_RUN_PROXY(TASK_ID=0) lazy                                                                                                                                                                                      
        [1]SIMPLE_PROJECT lazy                                                                                                                                                                                                  
          [0]REQUEST (cache_enable)                                                                                                                                                                                             
            ...                                                                                                                                                                                                                 
    [13]REQUEST_RUN_PROXY(TASK_ID=2) lazy                                                                                                                                                                                       
      [1]SIMPLE_PROJECT lazy                                                                                                                                                                                                    
        ...   
```